src/lingrex/trimming.py

```diff
@@ -0,0 +1,201 @@
+    """
+    Trim alignment sites by gaps.
+
+    :parameter alms: Alignment sites of a cognate set.
+    :type alms: list
+    :parameter threshold: Threshold by which sites with gaps should be trimmed.
+        Defaults to '0.5'.
+    :type threshold: int
+    :param skeletons: Tuple of syllable-skeletons that should be preserved
+        for further processing. Defaults to '("CV", "VC")'.
+    :type skeletons: tuple
+    :parameter gap: String that codes gaps in alignment sites. Defaults to '-'.
+    :type gap: string
+    :param exclude: Sequence of strings that should be excluded from further processing,
+        e.g. morpheme boundaries. Defaults to '_+'.
+    :param exclude: str
+    :return: Indices of trimmed strings.
+    :rtype: set
+    """
+    skeleton = get_skeleton(alms, gap=gap)
+    profile = gap_profile(alms, gap)
+    # exclude markers
+    idxs = [i for i, c in enumerate(skeleton) if c in exclude]
+    # order by gap weights
+    sorted_profile = sorted(enumerate(profile), key=lambda x: x[1], reverse=True)
+    while sorted_profile:
+        idx, score = sorted_profile.pop(0)
+        if score >= threshold:
+            current_skeleton = "".join([skeleton[i] for i in
+                                        range(len(skeleton)) if i not in idxs+[idx]])
+            if any([subsequence_of(s, current_skeleton) for s in skeletons]):
+                idxs += [idx]
+            else:
+                break
+        else:
+            break
+    return sorted(set(idxs))
+
+
+def trim_by_core(
+        alms, threshold=0.5, skeletons=("CV", "VC"), gap="-",
+        exclude="_+"
+        ):
+    """
+    Trim alignment sites by gaps, preserving a core of sites.
+
+    :parameter alms: Alignment sites of a cognate set.
+    :type alms: list
+    :parameter threshold: Threshold by which sites with gaps should be trimmed.
+        Defaults to '0.5'.
+    :type threshold: int
+    :param skeletons: Tuple of syllable-skeletons that should be preserved
+        for further processing. Defaults to '("CV", "VC")'.
+    :type skeletons: tuple
+    :parameter gap: String that codes gaps in alignment sites. Defaults to '-'.
+    :type gap: string
+    :return: Indices of trimmed strings.
+    :rtype: set
+    """
+    cons = [gap if c >= threshold else "S" for c in gap_profile(alms, gap=gap)]
+    skeleton = get_skeleton(alms, gap=gap)
+    left, right = consecutive_gaps(cons)
+
+    idxs = [i for i, c in enumerate(skeleton) if c in exclude]
+
+    # order by first
+    sorted_indices = right[::-1] + left
+    while sorted_indices:
+        idx = sorted_indices.pop(0)
+        current_skeleton = "".join([skeleton[i] for i in
+                                    range(len(skeleton)) if i not in idxs+[idx]])
+        if any([subsequence_of(s, current_skeleton) for s in skeletons]):
+            idxs += [idx]
+        else:
+            break
+    return sorted(set(idxs))
+
+
+def trim_random(
+        alms,
+        func=None,
```

Comment:
I think for complex arguments such as functions, type hinting is really useful. E.g.
```python
func=typing.Optional[typing.Callable[[...args...], typing.List[...]]],
```
See https://docs.python.org/3/library/typing.html#typing.Callable

Reply:
@tarotis, did you use type hints so far? If it is not clear, let us know here.

Reply:
Nop, I have not used type hints before I think